packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -99,6 +118,259 @@ class ReactShallowRenderer {
     this._rendering = false;
     this._forcedUpdate = false;
     this._updater = new Updater(this);
+    if (enableHooks) {
+      this._dispatcher = ReactCurrentDispatcher.current = this._createDispatcher();
```

Comment:
nit: Should we only set the `ReactCurrentDispatcher.current` to a non-null dispatcher when actually rendering? Not sure if this is strictly necessary, but it seems more semantically correct (and is more inline with other renderers).

Reply:
This sounds like a race condition to me 😓

Reply:
> This sounds like a race condition to me

What makes you say this?

Temporarily overriding the current dispatcher is a pattern we follow in several places. As long as   each place that overrides is careful to restore on completion, what's the race?